src/vsr/replica_format.zig

```diff
@@ -128,8 +132,32 @@ fn ReplicaFormatType(comptime Storage: type) type {
             assert(format_wal_headers(cluster, wal_offset, wal_buffer) == 0);
         }
 
-        fn format_wal_sectors_callback(write: *Storage.Write) void {
-            const self = @fieldParentPtr(Self, "wal_write", write);
+        fn format_replies(
+            self: *Self,
+            allocator: std.mem.Allocator,
+            storage: *Storage,
+        ) !void {
+            // Direct I/O requires the buffer to be sector-aligned.
+            var message_buffer =
+                try allocator.alignedAlloc(u8, constants.sector_size, constants.message_size_max);
+            defer allocator.free(message_buffer);
+            @memset(message_buffer, 0);
+
+            for (0..constants.clients_max) |slot| {
+                storage.write_sectors(
+                    write_sectors_callback,
+                    &self.write,
+                    message_buffer,
+                    .client_replies,
+                    slot * constants.message_size_max,
+                );
+                self.formatting = true;
+                while (self.formatting) storage.tick();
```

Comment:
unrelated to PR, but should we parallelize formatting? As in, I think we could perhaps issue more than one write at a time? 

We trust that parallel IO is safe during normal operation, so I don't think we get any extra reliability here by formatting sequentially? I guess this also ties into the idea about randomizing formatting: we can create a list of all formatting operations we want to do, and then execute them in parallel, in random order. 

formatting is going to be a bottleneck for at least some workflows, so making it faster might be meaningful.

Reply:
(implementation wise, I guess this means changing `formatting: bool` to a counter and just issuing several ops?)

Reply:
:+1: Added a TODO.